lib.typ

```diff
@@ -51,28 +52,56 @@ show ref: set text(fill: link-color)
 
 // ------------------- Math equation settings -------------------
 
-set math.equation(numbering: numbering-pattern) if equate-settings != none
-// only labeled equations get a number
+// either use equate if equate-settings is set or use i-figured if equate-settings is none
+// i-figured settings
 show math.equation: it => {
+  if equate-settings == none {
+    show-equation(prefix: "eq:", only-labeled: true, numbering: equation-numbering-pattern, it)
+  } else {
+    it
+  }
+}
+set math.equation(supplement: none) if equate-settings == none
+
+// equate settings
+show: it => {
   if equate-settings != none {
     equate(..equate-settings, it)
-  } else if it.has("label"){
-    math.equation(block:true, numbering: numbering-pattern, it)
   } else {
     it
   }
 }
+set math.equation(numbering: equation-numbering-pattern) if equate-settings != none
+
+// Reference equations with parentheses (for equate)
+// cf. https://forum.typst.app/t/how-can-i-set-numbering-for-sub-equations/1603/4
```

Comment:
Love it, that was something I also tried by myself but couldn't make it work yet!